File: Task 4/C++Stock Viewer/Product/Electronics.cpp
#include "Electronics.h"

Electronics::Electronics() : Product(){
    manufacturer = "";
}

Electronics::Electronics(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string manufacturer) 
            : Product(name, cost, supplier, discount_amount, packing_units, cost_per_unit){
    this->manufacturer = manufacturer;
}

Electronics::Electronics(const Electronics& other) : Product(other) {
    this->manufacturer = other.manufacturer;
}


std::string Electronics::display_product_info(){
    return Product::display_product_info() + 
            "\nManufacturer: " +manufacturer;
};

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Electronics.java
package com.logistics.Components.Product;

public class Electronics extends Product {

    protected String manufacturer;

    public Electronics() {
        super();
        manufacturer = "";
    }

    public Electronics(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String manufacturer) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit);
        this.manufacturer = manufacturer;
    }

    // Copy Constructor
    public Electronics(Electronics other) {
        super(other);
        this.manufacturer = other.manufacturer;
    }

    // Clone method
    @Override
    public Electronics clone() {
        return new Electronics(this);
    }

    // Getters
    public String getManufacturer() { return manufacturer; }

    // Setters
    public void setManufacturer(String manufacturer) { this.manufacturer = manufacturer; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nManufacturer: " + manufacturer;
    }
}
